// retell-sdk/example.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const { Language } = require("./models/components");
const sdk_1 = require("./sdk");
// import { CallType } from "./models/components";
// import { CallType, SortOrder } from "./models/operations";
const retell = new sdk_1.RetellClient({
    // Find the key in dashboard
    apiKey: "1a698b4f-5b06-4c71-8f69-7c506a67696e",
    serverURL: "http://localhost:3002"
});
async function main() {
    // const agents = await retell.listAgents();
    const res = await retell.createAgent({
      agentName: "testfunction",
      voiceId: "11labs-Ryan",
      llmWebsocketUrl: "",
      voiceTemperature: 2,
      enableBackchannel: false,
      webhookUrl: "test",
      voiceSpeed: 1,
      activateHippaCompliance: false,
      language: Language.DeDE,
      formatText: false,
    });

    
    // const agent = await retell.updateAgent({
    //     agentName: "new name",
    //     voiceTemperature: 0,
    //     enableBackchannel: true,
    //     webhookUrl: "test2",
    //     voiceSpeed: 1,
    //     boosted_keywords: ["test1"],
    //     activateHippaCompliance: false,
    //     language: Language.EnGB,
    //   formatText: true,
    // }, res.agent.agentId);
    // console.log(agent.agent);

    let res2 = await retell.registerCall({
      agentId: res.agent.agentId,
      audioWebsocketProtocol: "web",
      audioEncoding: "s16le",
      sampleRate: 22050,
      fromNumber: "+1234",
      toNumber: "+1234",
      metadata: {"meta":"meta"}
    });
     console.log(res2.callDetail);
    // const res2 = await retell.getAgent("68ebe15b1c323aa536bedfc275e02b53");
    // // const res = await retell.listAgents();
    // console.log(res2);

    // const res = await retell.deleteAgent('5790c196cd6b7728de819c3de444d3b5');
    // console.log(res.statusCode)
    // const res = await retell.createPhoneNumber({
    //     agentId: "oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD",
    //     areaCode: 415,
    // });
    // res.phoneNumber
    // const res = await retell.deletePhoneNumber("+14159095857");
    // console.log(res.statusCode)
    // const res = await retell.getCall("e208e53fe334b42a21bdf065f5b121e1");
    // console.log(res.callDetail);
    // const newPhoneNumber = await retell.createPhoneNumber({
    //     agentId: "5790c196cd6b7728de819c3de444d3b5", // Required
    //     // areaCode: 415, // Optional, default is 213
    //   });
    //   console.log("phone number: ", newPhoneNumber);
    // const ongoingCall = await retell.createPhoneCall({
    //     phoneNumber: {
    //       from: "+12138954697", // Phone number you created
    //       to: "+12137134981",
    //     },
    //   });
    //   console.log("phone call: ", ongoingCall);
    // const filterCriteria = {
    //     // afterEndTimestamp: 1703302428800,
    //     // afterStartTimestamp: 1703302407300,
    //     // agentId: ["oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD"],
    //     // beforeEndTimestamp: 1703302428899,
    //     // beforeStartTimestamp: 1703302407399,
    //     callType: [CallType.InboundPhoneCall, CallType.OutboundPhoneCall],
    // };
    // const limit = 1;
    // const sortOrder = SortOrder.Descending;
    // const res = await retell.listCalls(filterCriteria, limit, sortOrder);
    // console.log(res)
}
main();
