// retell-sdk/models/operations/createphonecall.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePhoneCallResponse$ = exports.CreatePhoneCallResponseBody$ = exports.CreatePhoneCallCallType$ = exports.CallStatus$ = exports.CreatePhoneCallRequestBody$ = exports.PhoneNumber$ = exports.CreatePhoneCallCallType = exports.CallStatus = void 0;
const components = __importStar(require("../../models/components"));
const zod_1 = require("zod");
/**
 * Status of call.
 */
var CallStatus;
(function (CallStatus) {
    CallStatus["Ongoing"] = "ongoing";
    CallStatus["Ended"] = "ended";
    CallStatus["Error"] = "error";
})(CallStatus || (exports.CallStatus = CallStatus = {}));
/**
 * Web call or phone call.
 */
var CreatePhoneCallCallType;
(function (CreatePhoneCallCallType) {
    CreatePhoneCallCallType["InboundPhoneCall"] = "inbound_phone_call";
    CreatePhoneCallCallType["OutboundPhoneCall"] = "outbound_phone_call";
    CreatePhoneCallCallType["WebCall"] = "web_call";
})(CreatePhoneCallCallType || (exports.CreatePhoneCallCallType = CreatePhoneCallCallType = {}));
/** @internal */
var PhoneNumber$;
(function (PhoneNumber$) {
    PhoneNumber$.inboundSchema = zod_1.z
        .object({
        from: zod_1.z.string(),
        to: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            from: v.from,
            to: v.to,
        };
    });
    PhoneNumber$.outboundSchema = zod_1.z
        .object({
        from: zod_1.z.string(),
        to: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            from: v.from,
            to: v.to,
        };
    });
})(PhoneNumber$ || (exports.PhoneNumber$ = PhoneNumber$ = {}));
/** @internal */
var CreatePhoneCallRequestBody$;
(function (CreatePhoneCallRequestBody$) {
    CreatePhoneCallRequestBody$.inboundSchema = zod_1.z
        .object({
        agent_prompt_params: zod_1.z
            .array(components.AgentPromptParams$.inboundSchema)
            .optional(),
        phone_number: zod_1.z.lazy(() => PhoneNumber$.inboundSchema),
    })
        .transform((v) => {
        return {
            ...(v.agent_prompt_params === undefined
                ? null
                : { agentPromptParams: v.agent_prompt_params }),
            phoneNumber: v.phone_number,
        };
    });
    CreatePhoneCallRequestBody$.outboundSchema = zod_1.z
        .object({
        agentPromptParams: zod_1.z
            .array(components.AgentPromptParams$.outboundSchema)
            .optional(),
        phoneNumber: zod_1.z.lazy(() => PhoneNumber$.outboundSchema),
    })
        .transform((v) => {
        return {
            ...(v.agentPromptParams === undefined
                ? null
                : { agent_prompt_params: v.agentPromptParams }),
            phone_number: v.phoneNumber,
        };
    });
})(CreatePhoneCallRequestBody$ || (exports.CreatePhoneCallRequestBody$ = CreatePhoneCallRequestBody$ = {}));
/** @internal */
exports.CallStatus$ = zod_1.z.nativeEnum(CallStatus);
/** @internal */
exports.CreatePhoneCallCallType$ = zod_1.z.nativeEnum(CreatePhoneCallCallType);
/** @internal */
var CreatePhoneCallResponseBody$;
(function (CreatePhoneCallResponseBody$) {
    CreatePhoneCallResponseBody$.inboundSchema = zod_1.z
        .object({
        agent_id: zod_1.z.string(),
        agent_prompt_params: zod_1.z
            .array(components.AgentPromptParams$.inboundSchema)
            .optional(),
        call_id: zod_1.z.string(),
        call_status: exports.CallStatus$,
        call_type: exports.CreatePhoneCallCallType$,
        phone_number: components.CallPhoneNumber$.inboundSchema,
        start_timestamp: zod_1.z.number().int(),
    })
        .transform((v) => {
        return {
            agentId: v.agent_id,
            ...(v.agent_prompt_params === undefined
                ? null
                : { agentPromptParams: v.agent_prompt_params }),
            callId: v.call_id,
            callStatus: v.call_status,
            callType: v.call_type,
            phoneNumber: v.phone_number,
            startTimestamp: v.start_timestamp,
        };
    });
    CreatePhoneCallResponseBody$.outboundSchema = zod_1.z
        .object({
        agentId: zod_1.z.string(),
        agentPromptParams: zod_1.z
            .array(components.AgentPromptParams$.outboundSchema)
            .optional(),
        callId: zod_1.z.string(),
        callStatus: exports.CallStatus$,
        callType: exports.CreatePhoneCallCallType$,
        phoneNumber: components.CallPhoneNumber$.outboundSchema,
        startTimestamp: zod_1.z.number().int(),
    })
        .transform((v) => {
        return {
            agent_id: v.agentId,
            ...(v.agentPromptParams === undefined
                ? null
                : { agent_prompt_params: v.agentPromptParams }),
            call_id: v.callId,
            call_status: v.callStatus,
            call_type: v.callType,
            phone_number: v.phoneNumber,
            start_timestamp: v.startTimestamp,
        };
    });
})(CreatePhoneCallResponseBody$ || (exports.CreatePhoneCallResponseBody$ = CreatePhoneCallResponseBody$ = {}));
/** @internal */
var CreatePhoneCallResponse$;
(function (CreatePhoneCallResponse$) {
    CreatePhoneCallResponse$.inboundSchema = zod_1.z
        .object({
        ContentType: zod_1.z.string(),
        StatusCode: zod_1.z.number().int(),
        RawResponse: zod_1.z.instanceof(Response),
        callDetail: zod_1.z
            .lazy(() => CreatePhoneCallResponseBody$.inboundSchema)
            .optional(),
    })
        .transform((v) => {
        return {
            contentType: v.ContentType,
            statusCode: v.StatusCode,
            rawResponse: v.RawResponse,
            ...(v.callDetail === undefined ? null : { callDetail: v.callDetail }),
        };
    });
    CreatePhoneCallResponse$.outboundSchema = zod_1.z
        .object({
        contentType: zod_1.z.string(),
        statusCode: zod_1.z.number().int(),
        rawResponse: zod_1.z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
        callDetail: zod_1.z
            .lazy(() => CreatePhoneCallResponseBody$.outboundSchema)
            .optional(),
    })
        .transform((v) => {
        return {
            ContentType: v.contentType,
            StatusCode: v.statusCode,
            RawResponse: v.rawResponse,
            ...(v.callDetail === undefined ? null : { callDetail: v.callDetail }),
        };
    });
})(CreatePhoneCallResponse$ || (exports.CreatePhoneCallResponse$ = CreatePhoneCallResponse$ = {}));
