// retell-sdk/models/operations/deletephonenumber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeletePhoneNumberResponse$ = exports.DeletePhoneNumberRequest$ = void 0;
const zod_1 = require("zod");
/** @internal */
var DeletePhoneNumberRequest$;
(function (DeletePhoneNumberRequest$) {
    DeletePhoneNumberRequest$.inboundSchema = zod_1.z
        .object({
        phone_number: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            phoneNumber: v.phone_number,
        };
    });
    DeletePhoneNumberRequest$.outboundSchema = zod_1.z
        .object({
        phoneNumber: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            phone_number: v.phoneNumber,
        };
    });
})(DeletePhoneNumberRequest$ || (exports.DeletePhoneNumberRequest$ = DeletePhoneNumberRequest$ = {}));
/** @internal */
var DeletePhoneNumberResponse$;
(function (DeletePhoneNumberResponse$) {
    DeletePhoneNumberResponse$.inboundSchema = zod_1.z
        .object({
        ContentType: zod_1.z.string(),
        StatusCode: zod_1.z.number().int(),
        RawResponse: zod_1.z.instanceof(Response),
    })
        .transform((v) => {
        return {
            contentType: v.ContentType,
            statusCode: v.StatusCode,
            rawResponse: v.RawResponse,
        };
    });
    DeletePhoneNumberResponse$.outboundSchema = zod_1.z
        .object({
        contentType: zod_1.z.string(),
        statusCode: zod_1.z.number().int(),
        rawResponse: zod_1.z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
    })
        .transform((v) => {
        return {
            ContentType: v.contentType,
            StatusCode: v.statusCode,
            RawResponse: v.rawResponse,
        };
    });
})(DeletePhoneNumberResponse$ || (exports.DeletePhoneNumberResponse$ = DeletePhoneNumberResponse$ = {}));
