// retell-sdk/models/operations/deleteagent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteAgentResponse$ = exports.DeleteAgentRequest$ = void 0;
const zod_1 = require("zod");
/** @internal */
var DeleteAgentRequest$;
(function (DeleteAgentRequest$) {
    DeleteAgentRequest$.inboundSchema = zod_1.z
        .object({
        agent_id: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            agentId: v.agent_id,
        };
    });
    DeleteAgentRequest$.outboundSchema = zod_1.z
        .object({
        agentId: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            agent_id: v.agentId,
        };
    });
})(DeleteAgentRequest$ || (exports.DeleteAgentRequest$ = DeleteAgentRequest$ = {}));
/** @internal */
var DeleteAgentResponse$;
(function (DeleteAgentResponse$) {
    DeleteAgentResponse$.inboundSchema = zod_1.z
        .object({
        ContentType: zod_1.z.string(),
        StatusCode: zod_1.z.number().int(),
        RawResponse: zod_1.z.instanceof(Response),
    })
        .transform((v) => {
        return {
            contentType: v.ContentType,
            statusCode: v.StatusCode,
            rawResponse: v.RawResponse,
        };
    });
    DeleteAgentResponse$.outboundSchema = zod_1.z
        .object({
        contentType: zod_1.z.string(),
        statusCode: zod_1.z.number().int(),
        rawResponse: zod_1.z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
    })
        .transform((v) => {
        return {
            ContentType: v.contentType,
            StatusCode: v.statusCode,
            RawResponse: v.rawResponse,
        };
    });
})(DeleteAgentResponse$ || (exports.DeleteAgentResponse$ = DeleteAgentResponse$ = {}));
