// retell-sdk/models/operations/getcall.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetCallResponse$ = exports.GetCallRequest$ = void 0;
const components = __importStar(require("../../models/components"));
const zod_1 = require("zod");
/** @internal */
var GetCallRequest$;
(function (GetCallRequest$) {
    GetCallRequest$.inboundSchema = zod_1.z
        .object({
        call_id: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            callId: v.call_id,
        };
    });
    GetCallRequest$.outboundSchema = zod_1.z
        .object({
        callId: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            call_id: v.callId,
        };
    });
})(GetCallRequest$ || (exports.GetCallRequest$ = GetCallRequest$ = {}));
/** @internal */
var GetCallResponse$;
(function (GetCallResponse$) {
    GetCallResponse$.inboundSchema = zod_1.z
        .object({
        CallDetail: components.CallDetail$.inboundSchema.optional(),
        ContentType: zod_1.z.string(),
        StatusCode: zod_1.z.number().int(),
        RawResponse: zod_1.z.instanceof(Response),
    })
        .transform((v) => {
        return {
            ...(v.CallDetail === undefined ? null : { callDetail: v.CallDetail }),
            contentType: v.ContentType,
            statusCode: v.StatusCode,
            rawResponse: v.RawResponse,
        };
    });
    GetCallResponse$.outboundSchema = zod_1.z
        .object({
        callDetail: components.CallDetail$.outboundSchema.optional(),
        contentType: zod_1.z.string(),
        statusCode: zod_1.z.number().int(),
        rawResponse: zod_1.z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
    })
        .transform((v) => {
        return {
            ...(v.callDetail === undefined ? null : { CallDetail: v.callDetail }),
            ContentType: v.contentType,
            StatusCode: v.statusCode,
            RawResponse: v.rawResponse,
        };
    });
})(GetCallResponse$ || (exports.GetCallResponse$ = GetCallResponse$ = {}));
