// retell-sdk/models/operations/getphonenumber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetPhoneNumberResponse$ = exports.GetPhoneNumberRequest$ = void 0;
const components = __importStar(require("../../models/components"));
const zod_1 = require("zod");
/** @internal */
var GetPhoneNumberRequest$;
(function (GetPhoneNumberRequest$) {
    GetPhoneNumberRequest$.inboundSchema = zod_1.z
        .object({
        phone_number: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            phoneNumber: v.phone_number,
        };
    });
    GetPhoneNumberRequest$.outboundSchema = zod_1.z
        .object({
        phoneNumber: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            phone_number: v.phoneNumber,
        };
    });
})(GetPhoneNumberRequest$ || (exports.GetPhoneNumberRequest$ = GetPhoneNumberRequest$ = {}));
/** @internal */
var GetPhoneNumberResponse$;
(function (GetPhoneNumberResponse$) {
    GetPhoneNumberResponse$.inboundSchema = zod_1.z
        .object({
        ContentType: zod_1.z.string(),
        PhoneNumber: components.PhoneNumber$.inboundSchema.optional(),
        StatusCode: zod_1.z.number().int(),
        RawResponse: zod_1.z.instanceof(Response),
    })
        .transform((v) => {
        return {
            contentType: v.ContentType,
            ...(v.PhoneNumber === undefined
                ? null
                : { phoneNumber: v.PhoneNumber }),
            statusCode: v.StatusCode,
            rawResponse: v.RawResponse,
        };
    });
    GetPhoneNumberResponse$.outboundSchema = zod_1.z
        .object({
        contentType: zod_1.z.string(),
        phoneNumber: components.PhoneNumber$.outboundSchema.optional(),
        statusCode: zod_1.z.number().int(),
        rawResponse: zod_1.z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
    })
        .transform((v) => {
        return {
            ContentType: v.contentType,
            ...(v.phoneNumber === undefined
                ? null
                : { PhoneNumber: v.phoneNumber }),
            StatusCode: v.statusCode,
            RawResponse: v.rawResponse,
        };
    });
})(GetPhoneNumberResponse$ || (exports.GetPhoneNumberResponse$ = GetPhoneNumberResponse$ = {}));
