// retell-sdk/models/operations/listcalls.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListCallsResponse$ = exports.ListCallsRequest$ = exports.SortOrder$ = exports.FilterCriteria$ = exports.CallType$ = exports.SortOrder = exports.CallType = void 0;
const components = __importStar(require("../../models/components"));
const zod_1 = require("zod");
var CallType;
(function (CallType) {
    CallType["InboundPhoneCall"] = "inbound_phone_call";
    CallType["OutboundPhoneCall"] = "outbound_phone_call";
    CallType["WebCall"] = "web_call";
})(CallType || (exports.CallType = CallType = {}));
/**
 * The calls will be sorted by `start_timestamp`, whether to return the calls in ascending or descending order.
 */
var SortOrder;
(function (SortOrder) {
    SortOrder["Ascending"] = "ascending";
    SortOrder["Descending"] = "descending";
})(SortOrder || (exports.SortOrder = SortOrder = {}));
/** @internal */
exports.CallType$ = zod_1.z.nativeEnum(CallType);
/** @internal */
var FilterCriteria$;
(function (FilterCriteria$) {
    FilterCriteria$.inboundSchema = zod_1.z
        .object({
        after_end_timestamp: zod_1.z.number().int().optional(),
        after_start_timestamp: zod_1.z.number().int().optional(),
        agent_id: zod_1.z.array(zod_1.z.string()).optional(),
        before_end_timestamp: zod_1.z.number().int().optional(),
        before_start_timestamp: zod_1.z.number().int().optional(),
    })
        .transform((v) => {
        return {
            ...(v.after_end_timestamp === undefined
                ? null
                : { afterEndTimestamp: v.after_end_timestamp }),
            ...(v.after_start_timestamp === undefined
                ? null
                : { afterStartTimestamp: v.after_start_timestamp }),
            ...(v.agent_id === undefined ? null : { agentId: v.agent_id }),
            ...(v.before_end_timestamp === undefined
                ? null
                : { beforeEndTimestamp: v.before_end_timestamp }),
            ...(v.before_start_timestamp === undefined
                ? null
                : { beforeStartTimestamp: v.before_start_timestamp }),
        };
    });
    FilterCriteria$.outboundSchema = zod_1.z
        .object({
        afterEndTimestamp: zod_1.z.number().int().optional(),
        afterStartTimestamp: zod_1.z.number().int().optional(),
        agentId: zod_1.z.array(zod_1.z.string()).optional(),
        beforeEndTimestamp: zod_1.z.number().int().optional(),
        beforeStartTimestamp: zod_1.z.number().int().optional(),
    })
        .transform((v) => {
        return {
            ...(v.afterEndTimestamp === undefined
                ? null
                : { after_end_timestamp: v.afterEndTimestamp }),
            ...(v.afterStartTimestamp === undefined
                ? null
                : { after_start_timestamp: v.afterStartTimestamp }),
            ...(v.agentId === undefined ? null : { agent_id: v.agentId }),
            ...(v.beforeEndTimestamp === undefined
                ? null
                : { before_end_timestamp: v.beforeEndTimestamp }),
            ...(v.beforeStartTimestamp === undefined
                ? null
                : { before_start_timestamp: v.beforeStartTimestamp }),
        };
    });
})(FilterCriteria$ || (exports.FilterCriteria$ = FilterCriteria$ = {}));
/** @internal */
exports.SortOrder$ = zod_1.z.nativeEnum(SortOrder);
/** @internal */
var ListCallsRequest$;
(function (ListCallsRequest$) {
    ListCallsRequest$.inboundSchema = zod_1.z
        .object({
        filter_criteria: zod_1.z.lazy(() => FilterCriteria$.inboundSchema).optional(),
        limit: zod_1.z.number().int().optional(),
        sort_order: exports.SortOrder$.optional(),
    })
        .transform((v) => {
        return {
            ...(v.filter_criteria === undefined
                ? null
                : { filterCriteria: v.filter_criteria }),
            ...(v.limit === undefined ? null : { limit: v.limit }),
            ...(v.sort_order === undefined ? null : { sortOrder: v.sort_order }),
        };
    });
    ListCallsRequest$.outboundSchema = zod_1.z
        .object({
        filterCriteria: zod_1.z.lazy(() => FilterCriteria$.outboundSchema).optional(),
        limit: zod_1.z.number().int().optional(),
        sortOrder: exports.SortOrder$.optional(),
    })
        .transform((v) => {
        return {
            ...(v.filterCriteria === undefined
                ? null
                : { filter_criteria: v.filterCriteria }),
            ...(v.limit === undefined ? null : { limit: v.limit }),
            ...(v.sortOrder === undefined ? null : { sort_order: v.sortOrder }),
        };
    });
})(ListCallsRequest$ || (exports.ListCallsRequest$ = ListCallsRequest$ = {}));
/** @internal */
var ListCallsResponse$;
(function (ListCallsResponse$) {
    ListCallsResponse$.inboundSchema = zod_1.z
        .object({
        ContentType: zod_1.z.string(),
        StatusCode: zod_1.z.number().int(),
        RawResponse: zod_1.z.instanceof(Response),
        calls: zod_1.z.array(components.CallDetail$.inboundSchema).optional(),
    })
        .transform((v) => {
        return {
            contentType: v.ContentType,
            statusCode: v.StatusCode,
            rawResponse: v.RawResponse,
            ...(v.calls === undefined ? null : { calls: v.calls }),
        };
    });
    ListCallsResponse$.outboundSchema = zod_1.z
        .object({
        contentType: zod_1.z.string(),
        statusCode: zod_1.z.number().int(),
        rawResponse: zod_1.z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
        calls: zod_1.z.array(components.CallDetail$.outboundSchema).optional(),
    })
        .transform((v) => {
        return {
            ContentType: v.contentType,
            StatusCode: v.statusCode,
            RawResponse: v.rawResponse,
            ...(v.calls === undefined ? null : { calls: v.calls }),
        };
    });
})(ListCallsResponse$ || (exports.ListCallsResponse$ = ListCallsResponse$ = {}));
