// retell-sdk/models/operations/registercall.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegisterCallResponse$ = exports.RegisterCallRequestBody$ = void 0;
const components_1 = require("../../models/components");
const components = __importStar(require("../../models/components"));
const zod_1 = require("zod");
/** @internal */
var RegisterCallRequestBody$;
(function (RegisterCallRequestBody$) {
    RegisterCallRequestBody$.inboundSchema = zod_1.z
        .object({
        agent_id: zod_1.z.string(),
        call_id: zod_1.z.string(),
        audio_encoding: components_1.AudioEncoding$,
        sample_rate: zod_1.z.number().int(),
        audio_websocket_protocol: components_1.AudioWebsocketProtocol$,
        end_call_after_silence_ms: zod_1.z.number().int().optional(),
        from_number: zod_1.z.string().optional(),
        to_number: zod_1.z.string().optional(),
        metadata: zod_1.z.record(zod_1.z.any()).optional(),
    })
        .transform((v) => {
        return {
            agentId: v.agent_id,
            audioEncoding: v.audio_encoding,
            sampleRate: v.sample_rate,
            audioWebsocketProtocol: v.audio_websocket_protocol,
            ...(v.end_call_after_silence_ms === undefined ? null : { endCallAfterSilenceMs: v.end_call_after_silence_ms }),
            ...(v.from_number === undefined ? null : { fromNumber: v.from_number }),
            ...(v.to_number === undefined ? null : { toNumber: v.to_number }),
            ...(v.metadata === undefined ? null : { metadata: v.metadata }),
        };
    });
    RegisterCallRequestBody$.outboundSchema = zod_1.z
        .object({
        agentId: zod_1.z.string(),
        audioEncoding: components_1.AudioEncoding$,
        sampleRate: zod_1.z.number().int(),
        audioWebsocketProtocol: components_1.AudioWebsocketProtocol$,
        endCallAfterSilenceMs: zod_1.z.number().int().optional(),
        fromNumber: zod_1.z.string().optional(),
        toNumber: zod_1.z.string().optional(),
        metadata: zod_1.z.record(zod_1.z.any()).optional(),
    })
        .transform((v) => {
        return {
            agent_id: v.agentId,
            audio_encoding: v.audioEncoding,
            sample_rate: v.sampleRate,
            audio_websocket_protocol: v.audioWebsocketProtocol,
            ...(v.endCallAfterSilenceMs === undefined ? null : { end_call_after_silence_ms: v.endCallAfterSilenceMs }),
            ...(v.fromNumber === undefined ? null : { from_number: v.fromNumber }),
            ...(v.toNumber === undefined ? null : { to_number: v.toNumber }),
            ...(v.metadata === undefined ? null : { metadata: v.metadata }),
        };
    });
})(RegisterCallRequestBody$ || (exports.RegisterCallRequestBody$ = RegisterCallRequestBody$ = {}));
/** @internal */
var RegisterCallResponse$;
(function (RegisterCallResponse$) {
    RegisterCallResponse$.inboundSchema = zod_1.z
        .object({
        CallDetail: components.CallDetail$.inboundSchema.optional(),
        ContentType: zod_1.z.string(),
        StatusCode: zod_1.z.number().int(),
        RawResponse: zod_1.z.instanceof(Response),
    })
        .transform((v) => {
        return {
            ...(v.CallDetail === undefined ? null : { callDetail: v.CallDetail }),
            contentType: v.ContentType,
            statusCode: v.StatusCode,
            rawResponse: v.RawResponse,
        };
    });
    RegisterCallResponse$.outboundSchema = zod_1.z
        .object({
        callDetail: components.CallDetail$.outboundSchema.optional(),
        contentType: zod_1.z.string(),
        statusCode: zod_1.z.number().int(),
        rawResponse: zod_1.z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
    })
        .transform((v) => {
        return {
            ...(v.callDetail === undefined ? null : { CallDetail: v.callDetail }),
            ContentType: v.contentType,
            StatusCode: v.statusCode,
            RawResponse: v.rawResponse,
        };
    });
})(RegisterCallResponse$ || (exports.RegisterCallResponse$ = RegisterCallResponse$ = {}));
