// retell-sdk/models/operations/updateagent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAgentResponse$ = exports.UpdateAgentRequest$ = exports.UpdateAgentRequestBody$ = void 0;
const components = __importStar(require("../../models/components"));
const zod_1 = require("zod");
const components_1 = require("../../models/components");
/** @internal */
var UpdateAgentRequestBody$;
(function (UpdateAgentRequestBody$) {
    UpdateAgentRequestBody$.inboundSchema = zod_1.z
        .object({
        agent_name: zod_1.z.string().optional(),
        llm_websocket_url: zod_1.z.string(),
        voice_id: zod_1.z.string(),
        enable_backchannel: zod_1.z.boolean().optional(),
        voice_temperature: zod_1.z.number().optional(),
        voice_speed: zod_1.z.number().optional(),
        responsiveness: zod_1.z.number().optional(),
        ambient_sound: zod_1.z.enum(['coffee-shop', 'convention-hall', 'summer-outdoor', 'mountain-outdoor', 'null']).optional(),
        webhook_url: zod_1.z.string().optional(),
        boosted_keywords: zod_1.z.array(zod_1.z.string()).optional(),
        opt_out_sensitive_data_storage: zod_1.z.boolean().optional(),
        format_text: zod_1.z.boolean().optional(),
        language: zod_1.z.nativeEnum(components_1.Language).optional(),
    })
        .transform((v) => {
        return {
            ...(v.agent_name === undefined ? null : { agentName: v.agent_name }),
            ...(v.llm_websocket_url === undefined
                ? null
                : { llmWebsocketUrl: v.llm_websocket_url }),
            ...(v.voice_id === undefined ? null : { voiceId: v.voice_id }),
            ...(v.enable_backchannel === undefined ? null : { enableBackchannel: v.enable_backchannel }),
            ...(v.voice_temperature === undefined ? null : { voiceTemperature: v.voice_temperature }),
            ...(v.voice_temperature === undefined ? null : { voiceTemperature: v.voice_temperature }),
            ...(v.voice_speed === undefined ? null : { voiceSpeed: v.voice_speed }),
            ...(v.responsiveness === undefined ? null : { responsiveness: v.responsiveness }),
            ...(v.ambient_sound === undefined ? null : { ambientSound: v.ambient_sound }),
            ...(v.webhook_url === undefined ? null : { webhookUrl: v.webhook_url }),
            ...(v.boosted_keywords === undefined ? null : { boostedKeywords: v.boosted_keywords }),
            ...(v.opt_out_sensitive_data_storage === undefined ? null : { optOutSensitiveDataStorage: v.opt_out_sensitive_data_storage }),
            ...(v.format_text === undefined ? null : { formatText: v.format_text }),
            ...(v.language === undefined ? null : { language: v.language }),
        };
    });
    UpdateAgentRequestBody$.outboundSchema = zod_1.z
        .object({
        agentName: zod_1.z.string().optional(),
        llmWebsocketUrl: zod_1.z.string().optional(),
        voiceId: zod_1.z.string().optional(),
        enableBackchannel: zod_1.z.boolean().optional(),
        voiceTemperature: zod_1.z.number().optional(),
        voiceSpeed: zod_1.z.number().optional(),
        responsiveness: zod_1.z.number().optional(),
        ambientSound: zod_1.z.enum(['coffee-shop', 'convention-hall', 'summer-outdoor', 'mountain-outdoor', 'null']).optional(),
        webhookUrl: zod_1.z.string().optional(),
        boostedKeywords: zod_1.z.array(zod_1.z.string()).optional(),
        optOutSensitiveDataStorage: zod_1.z.boolean().optional(),
        formatText: zod_1.z.boolean().optional(),
        language: zod_1.z.nativeEnum(components_1.Language).optional(),
    })
        .transform((v) => {
        return {
            ...(v.agentName === undefined ? null : { agent_name: v.agentName }),
            ...(v.llmWebsocketUrl === undefined
                ? null
                : { llm_websocket_url: v.llmWebsocketUrl }),
            ...(v.voiceId === undefined ? null : { voice_id: v.voiceId }),
            ...(v.enableBackchannel === undefined ? null : { enable_backchannel: v.enableBackchannel }),
            ...(v.voiceTemperature === undefined ? null : { voice_temperature: v.voiceTemperature }),
            ...(v.voiceSpeed === undefined ? null : { voice_speed: v.voiceSpeed }),
            ...(v.responsiveness === undefined ? null : { responsiveness: v.responsiveness }),
            ...(v.ambientSound === undefined ? null : { ambient_sound: v.ambientSound }),
            ...(v.webhookUrl === undefined ? null : { webhook_url: v.webhookUrl }),
            ...(v.boostedKeywords === undefined ? null : { boosted_keywords: v.boostedKeywords }),
            ...(v.optOutSensitiveDataStorage === undefined ? null : { opt_out_sensitive_data_storage: v.optOutSensitiveDataStorage }),
            ...(v.formatText === undefined ? null : { format_text: v.formatText }),
            ...(v.language === undefined ? null : { language: v.language }),
        };
    });
})(UpdateAgentRequestBody$ || (exports.UpdateAgentRequestBody$ = UpdateAgentRequestBody$ = {}));
/** @internal */
var UpdateAgentRequest$;
(function (UpdateAgentRequest$) {
    UpdateAgentRequest$.inboundSchema = zod_1.z
        .object({
        RequestBody: UpdateAgentRequestBody$.inboundSchema,
        agent_id: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            requestBody: v.RequestBody,
            agentId: v.agent_id,
        };
    });
    UpdateAgentRequest$.outboundSchema = zod_1.z
        .object({
        requestBody: UpdateAgentRequestBody$.outboundSchema,
        agentId: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            RequestBody: v.requestBody,
            agent_id: v.agentId,
        };
    });
})(UpdateAgentRequest$ || (exports.UpdateAgentRequest$ = UpdateAgentRequest$ = {}));
/** @internal */
var UpdateAgentResponse$;
(function (UpdateAgentResponse$) {
    UpdateAgentResponse$.inboundSchema = zod_1.z
        .object({
        Agent: components.Agent$.inboundSchema.optional(),
        ContentType: zod_1.z.string(),
        StatusCode: zod_1.z.number().int(),
        RawResponse: zod_1.z.instanceof(Response),
    })
        .transform((v) => {
        return {
            ...(v.Agent === undefined ? null : { agent: v.Agent }),
            contentType: v.ContentType,
            statusCode: v.StatusCode,
            rawResponse: v.RawResponse,
        };
    });
    UpdateAgentResponse$.outboundSchema = zod_1.z
        .object({
        agent: components.Agent$.outboundSchema.optional(),
        contentType: zod_1.z.string(),
        statusCode: zod_1.z.number().int(),
        rawResponse: zod_1.z.instanceof(Response).transform(() => {
            throw new Error("Response cannot be serialized");
        }),
    })
        .transform((v) => {
        return {
            ...(v.agent === undefined ? null : { Agent: v.agent }),
            ContentType: v.contentType,
            StatusCode: v.statusCode,
            RawResponse: v.rawResponse,
        };
    });
})(UpdateAgentResponse$ || (exports.UpdateAgentResponse$ = UpdateAgentResponse$ = {}));
