// retell-sdk/lib/config.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SDK_METADATA = exports.serverURLFromOptions = exports.ServerList = void 0;
const url_1 = require("./url");
/**
 * Contains the list of servers available to the SDK
 */
exports.ServerList = [
    /**
     * The production server.
     */
    "https://api.retellai.com",
];
function serverURLFromOptions(options) {
    var _a;
    let serverURL = options.serverURL;
    const params = {};
    const serverIdx = (_a = options.serverIdx) !== null && _a !== void 0 ? _a : 0;
    if (!serverURL) {
        serverURL = exports.ServerList[serverIdx] || "";
    }
    const u = (0, url_1.pathToFunc)(serverURL)(params);
    return new URL(u);
}
exports.serverURLFromOptions = serverURLFromOptions;
exports.SDK_METADATA = Object.freeze({
    language: "typescript",
    openapiDocVersion: "1.0.0",
    sdkVersion: "1.15.9",
    genVersion: "2.228.1",
    userAgent: "speakeasy-sdk/typescript 1.15.9 2.228.1 1.0.0 retell-sdk",
});
