// retell-sdk/models/components/agent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Agent$ = exports.Language = void 0;
const zod_1 = require("zod");
var Language;
(function (Language) {
    Language["EnUS"] = "en-US";
    Language["EnIN"] = "en-IN";
    Language["EnGB"] = "en-GB";
    Language["DeDE"] = "de-DE";
    Language["EsES"] = "es-ES";
    Language["Es419"] = "es-419";
    Language["HiIN"] = "hi-IN";
    Language["JaJP"] = "ja-JP";
    Language["PtPT"] = "pt-PT";
    Language["PtBR"] = "pt-BR";
})(Language || (exports.Language = Language = {}));
/** @internal */
var Agent$;
(function (Agent$) {
    Agent$.inboundSchema = zod_1.z
        .object({
        agent_id: zod_1.z.string(),
        agent_name: zod_1.z.string().optional(),
        llm_websocket_url: zod_1.z.string(),
        voice_id: zod_1.z.string(),
        last_modification_timestamp: zod_1.z.number().int(),
        enable_backchannel: zod_1.z.boolean().optional(),
        voice_temperature: zod_1.z.number().optional(),
        voice_speed: zod_1.z.number().optional(),
        responsiveness: zod_1.z.number().optional(),
        ambient_sound: zod_1.z.enum(['coffee-shop', 'convention-hall', 'summer-outdoor', 'mountain-outdoor', 'null']).optional(),
        webhook_url: zod_1.z.string().optional(),
        boosted_keywords: zod_1.z.array(zod_1.z.string()).optional(),
        opt_out_sensitive_data_storage: zod_1.z.boolean().optional(),
        format_text: zod_1.z.boolean().optional(),
        language: zod_1.z.nativeEnum(Language).optional(),
    })
        .transform((v) => {
        return {
            agentId: v.agent_id,
            ...(v.agent_name === undefined ? null : { agentName: v.agent_name }),
            llmWebsocketUrl: v.llm_websocket_url,
            voiceId: v.voice_id,
            lastModificationTimestamp: v.last_modification_timestamp,
            ...(v.enable_backchannel === undefined ? null : { enableBackchannel: v.enable_backchannel }),
            ...(v.voice_temperature === undefined ? null : { voiceTemperature: v.voice_temperature }),
            ...(v.voice_speed === undefined ? null : { voiceSpeed: v.voice_speed }),
            ...(v.responsiveness === undefined ? null : { responsiveness: v.responsiveness }),
            ...(v.ambient_sound === undefined ? null : { ambientSound: v.ambient_sound }),
            ...(v.webhook_url === undefined ? null : { webhookUrl: v.webhook_url }),
            ...(v.boosted_keywords === undefined ? null : { boostedKeywords: v.boosted_keywords }),
            ...(v.opt_out_sensitive_data_storage === undefined ? null : { optOutSensitiveDataStorage: v.opt_out_sensitive_data_storage }),
            ...(v.format_text === undefined ? null : { formatText: v.format_text }),
            ...(v.language === undefined ? null : { language: v.language }),
        };
    });
    Agent$.outboundSchema = zod_1.z
        .object({
        agentId: zod_1.z.string(),
        agentName: zod_1.z.string().optional(),
        llmWebsocketUrl: zod_1.z.string(),
        voiceId: zod_1.z.string(),
        lastModificationTimestamp: zod_1.z.number().int(),
        enableBackchannel: zod_1.z.boolean().optional(),
        voiceTemperature: zod_1.z.number().optional(),
        voiceSpeed: zod_1.z.number().optional(),
        responsiveness: zod_1.z.number().optional(),
        ambientSound: zod_1.z.enum(['coffee-shop', 'convention-hall', 'summer-outdoor', 'mountain-outdoor', 'null']).optional(),
        webhookUrl: zod_1.z.string().optional(),
        boostedKeywords: zod_1.z.array(zod_1.z.string()).optional(),
        optOutSensitiveDataStorage: zod_1.z.boolean().optional(),
        formatText: zod_1.z.boolean().optional(),
        language: zod_1.z.nativeEnum(Language).optional(),
    })
        .transform((v) => {
        return {
            agent_id: v.agentId,
            ...(v.agentName === undefined ? null : { agent_name: v.agentName }),
            llm_websocket_url: v.llmWebsocketUrl,
            voice_id: v.voiceId,
            last_modification_timestamp: v.lastModificationTimestamp,
            ...(v.enableBackchannel === undefined ? null : { enable_backchannel: v.enableBackchannel }),
            ...(v.voiceTemperature === undefined ? null : { voice_temperature: v.voiceTemperature }),
            ...(v.voiceSpeed === undefined ? null : { voice_speed: v.voiceSpeed }),
            ...(v.responsiveness === undefined ? null : { responsiveness: v.responsiveness }),
            ...(v.ambientSound === undefined ? null : { ambient_sound: v.ambientSound }),
            ...(v.webhookUrl === undefined ? null : { webhook_url: v.webhookUrl }),
            ...(v.boostedKeywords === undefined ? null : { boosted_keywords: v.boostedKeywords }),
            ...(v.optOutSensitiveDataStorage === undefined ? null : { opt_out_sensitive_data_storage: v.optOutSensitiveDataStorage }),
            ...(v.formatText === undefined ? null : { format_text: v.formatText }),
            ...(v.language === undefined ? null : { language: v.language }),
        };
    });
})(Agent$ || (exports.Agent$ = Agent$ = {}));
