// retell-sdk/models/components/calldetail.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallDetail$ = exports.AudioEncoding$ = exports.AudioWebsocketProtocol$ = exports.CallStatus$ = exports.AudioWebsocketProtocol = exports.AudioEncoding = exports.CallStatus = void 0;
const zod_1 = require("zod");
/**
 * Status of call.
 */
var CallStatus;
(function (CallStatus) {
    CallStatus["Registered"] = "registered";
    CallStatus["Ongoing"] = "ongoing";
    CallStatus["Ended"] = "ended";
    CallStatus["Error"] = "error";
})(CallStatus || (exports.CallStatus = CallStatus = {}));
/**
 * The audio encoding of the call.
 */
var AudioEncoding;
(function (AudioEncoding) {
    AudioEncoding["S16le"] = "s16le";
    AudioEncoding["Mulaw"] = "mulaw";
})(AudioEncoding || (exports.AudioEncoding = AudioEncoding = {}));
/**
 * The protocol how audio websocket read and send audio bytes.
 */
var AudioWebsocketProtocol;
(function (AudioWebsocketProtocol) {
    AudioWebsocketProtocol["Web"] = "web";
    AudioWebsocketProtocol["Twilio"] = "twilio";
    AudioWebsocketProtocol["Vonage"] = "vonage";
})(AudioWebsocketProtocol || (exports.AudioWebsocketProtocol = AudioWebsocketProtocol = {}));
/** @internal */
exports.CallStatus$ = zod_1.z.nativeEnum(CallStatus);
/** @internal */
exports.AudioWebsocketProtocol$ = zod_1.z.nativeEnum(AudioWebsocketProtocol);
/** @internal */
exports.AudioEncoding$ = zod_1.z.nativeEnum(AudioEncoding);
/** @internal */
var CallDetail$;
(function (CallDetail$) {
    CallDetail$.inboundSchema = zod_1.z
        .object({
        agent_id: zod_1.z.string(),
        call_id: zod_1.z.string(),
        call_status: exports.CallStatus$,
        audio_encoding: exports.AudioEncoding$,
        sample_rate: zod_1.z.number().int(),
        audio_websocket_protocol: exports.AudioWebsocketProtocol$,
        end_timestamp: zod_1.z.number().int().optional(),
        recording_url: zod_1.z.string().optional(),
        start_timestamp: zod_1.z.number().int(),
        transcript: zod_1.z.string().optional(),
        end_call_after_silence_ms: zod_1.z.number().int().optional(),
        from_number: zod_1.z.string().optional(),
        to_number: zod_1.z.string().optional(),
        metadata: zod_1.z.record(zod_1.z.any()).optional(),
    })
        .transform((v) => {
        return {
            agentId: v.agent_id,
            callId: v.call_id,
            callStatus: v.call_status,
            audioEncoding: v.audio_encoding,
            sampleRate: v.sample_rate,
            audioWebsocketProtocol: v.audio_websocket_protocol,
            ...(v.end_timestamp === undefined
                ? null
                : { endTimestamp: v.end_timestamp }),
            ...(v.recording_url === undefined
                ? null
                : { recordingUrl: v.recording_url }),
            startTimestamp: v.start_timestamp,
            ...(v.transcript === undefined ? null : { transcript: v.transcript }),
            ...(v.end_call_after_silence_ms === undefined ? null : { endCallAfterSilenceMs: v.end_call_after_silence_ms }),
            ...(v.from_number === undefined ? null : { fromNumber: v.from_number }),
            ...(v.to_number === undefined ? null : { toNumber: v.to_number }),
            ...(v.metadata === undefined ? null : { metadata: v.metadata }),
        };
    });
    CallDetail$.outboundSchema = zod_1.z
        .object({
        agentId: zod_1.z.string(),
        callId: zod_1.z.string(),
        callStatus: exports.CallStatus$,
        audioEncoding: exports.AudioEncoding$,
        sampleRate: zod_1.z.number().int(),
        audioWebsocketProtocol: exports.AudioWebsocketProtocol$,
        endTimestamp: zod_1.z.number().int().optional(),
        recordingUrl: zod_1.z.string().optional(),
        startTimestamp: zod_1.z.number().int(),
        transcript: zod_1.z.string().optional(),
        endCallAfterSilenceMs: zod_1.z.number().int().optional(),
        fromNumber: zod_1.z.string().optional(),
        toNumber: zod_1.z.string().optional(),
        metadata: zod_1.z.record(zod_1.z.any()).optional(),
    })
        .transform((v) => {
        return {
            agent_id: v.agentId,
            call_id: v.callId,
            call_status: v.callStatus,
            audio_encoding: v.audioEncoding,
            sample_rate: v.sampleRate,
            audio_websocket_protocol: v.audioWebsocketProtocol,
            ...(v.endTimestamp === undefined
                ? null
                : { end_timestamp: v.endTimestamp }),
            ...(v.recordingUrl === undefined
                ? null
                : { recording_url: v.recordingUrl }),
            start_timestamp: v.startTimestamp,
            ...(v.transcript === undefined ? null : { transcript: v.transcript }),
            ...(v.endCallAfterSilenceMs === undefined ? null : { end_call_after_silence_ms: v.endCallAfterSilenceMs }),
            ...(v.fromNumber === undefined ? null : { from_number: v.fromNumber }),
            ...(v.toNumber === undefined ? null : { to_number: v.toNumber }),
            ...(v.metadata === undefined ? null : { metadata: v.metadata }),
        };
    });
})(CallDetail$ || (exports.CallDetail$ = CallDetail$ = {}));
