// retell-sdk/models/components/phonenumber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhoneNumber$ = void 0;
const zod_1 = require("zod");
/** @internal */
var PhoneNumber$;
(function (PhoneNumber$) {
    PhoneNumber$.inboundSchema = zod_1.z
        .object({
        agent_id: zod_1.z.string(),
        area_code: zod_1.z.number().int(),
        last_modification_timestamp: zod_1.z.number().int(),
        phone_number: zod_1.z.string(),
        phone_number_pretty: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            agentId: v.agent_id,
            areaCode: v.area_code,
            lastModificationTimestamp: v.last_modification_timestamp,
            phoneNumber: v.phone_number,
            phoneNumberPretty: v.phone_number_pretty,
        };
    });
    PhoneNumber$.outboundSchema = zod_1.z
        .object({
        agentId: zod_1.z.string(),
        areaCode: zod_1.z.number().int(),
        lastModificationTimestamp: zod_1.z.number().int(),
        phoneNumber: zod_1.z.string(),
        phoneNumberPretty: zod_1.z.string(),
    })
        .transform((v) => {
        return {
            agent_id: v.agentId,
            area_code: v.areaCode,
            last_modification_timestamp: v.lastModificationTimestamp,
            phone_number: v.phoneNumber,
            phone_number_pretty: v.phoneNumberPretty,
        };
    });
})(PhoneNumber$ || (exports.PhoneNumber$ = PhoneNumber$ = {}));
