// retell-sdk/models/errors/createphonenumber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePhoneNumberResponseBody$ = exports.CreatePhoneNumberResponseResponseBody$ = exports.CreatePhoneNumberResponse402ResponseBody$ = exports.CreatePhoneNumberResponse422ResponseBody$ = exports.CreatePhoneNumberResponse500ResponseBody$ = exports.CreatePhoneNumberResponseBody = exports.CreatePhoneNumberResponseResponseBody = exports.CreatePhoneNumberResponse402ResponseBody = exports.CreatePhoneNumberResponse422ResponseBody = exports.CreatePhoneNumberResponse500ResponseBody = void 0;
const zod_1 = require("zod");
/**
 * Internal Server Error
 */
class CreatePhoneNumberResponse500ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreatePhoneNumberResponse500ResponseBody";
    }
}
exports.CreatePhoneNumberResponse500ResponseBody = CreatePhoneNumberResponse500ResponseBody;
/**
 * Unprocessable Content
 */
class CreatePhoneNumberResponse422ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreatePhoneNumberResponse422ResponseBody";
    }
}
exports.CreatePhoneNumberResponse422ResponseBody = CreatePhoneNumberResponse422ResponseBody;
/**
 * Payment Required
 */
class CreatePhoneNumberResponse402ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreatePhoneNumberResponse402ResponseBody";
    }
}
exports.CreatePhoneNumberResponse402ResponseBody = CreatePhoneNumberResponse402ResponseBody;
/**
 * Unauthorized
 */
class CreatePhoneNumberResponseResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreatePhoneNumberResponseResponseBody";
    }
}
exports.CreatePhoneNumberResponseResponseBody = CreatePhoneNumberResponseResponseBody;
/**
 * Bad Request
 */
class CreatePhoneNumberResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreatePhoneNumberResponseBody";
    }
}
exports.CreatePhoneNumberResponseBody = CreatePhoneNumberResponseBody;
/** @internal */
var CreatePhoneNumberResponse500ResponseBody$;
(function (CreatePhoneNumberResponse500ResponseBody$) {
    CreatePhoneNumberResponse500ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreatePhoneNumberResponse500ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreatePhoneNumberResponse500ResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreatePhoneNumberResponse500ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreatePhoneNumberResponse500ResponseBody$ || (exports.CreatePhoneNumberResponse500ResponseBody$ = CreatePhoneNumberResponse500ResponseBody$ = {}));
/** @internal */
var CreatePhoneNumberResponse422ResponseBody$;
(function (CreatePhoneNumberResponse422ResponseBody$) {
    CreatePhoneNumberResponse422ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreatePhoneNumberResponse422ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreatePhoneNumberResponse422ResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreatePhoneNumberResponse422ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreatePhoneNumberResponse422ResponseBody$ || (exports.CreatePhoneNumberResponse422ResponseBody$ = CreatePhoneNumberResponse422ResponseBody$ = {}));
/** @internal */
var CreatePhoneNumberResponse402ResponseBody$;
(function (CreatePhoneNumberResponse402ResponseBody$) {
    CreatePhoneNumberResponse402ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreatePhoneNumberResponse402ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreatePhoneNumberResponse402ResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreatePhoneNumberResponse402ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreatePhoneNumberResponse402ResponseBody$ || (exports.CreatePhoneNumberResponse402ResponseBody$ = CreatePhoneNumberResponse402ResponseBody$ = {}));
/** @internal */
var CreatePhoneNumberResponseResponseBody$;
(function (CreatePhoneNumberResponseResponseBody$) {
    CreatePhoneNumberResponseResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreatePhoneNumberResponseResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreatePhoneNumberResponseResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreatePhoneNumberResponseResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreatePhoneNumberResponseResponseBody$ || (exports.CreatePhoneNumberResponseResponseBody$ = CreatePhoneNumberResponseResponseBody$ = {}));
/** @internal */
var CreatePhoneNumberResponseBody$;
(function (CreatePhoneNumberResponseBody$) {
    CreatePhoneNumberResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreatePhoneNumberResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreatePhoneNumberResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreatePhoneNumberResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreatePhoneNumberResponseBody$ || (exports.CreatePhoneNumberResponseBody$ = CreatePhoneNumberResponseBody$ = {}));
