// retell-sdk/models/errors/createagent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAgentResponseBody$ = exports.CreateAgentResponseResponseBody$ = exports.CreateAgentResponse422ResponseBody$ = exports.CreateAgentResponse500ResponseBody$ = exports.CreateAgentResponseBody = exports.CreateAgentResponseResponseBody = exports.CreateAgentResponse422ResponseBody = exports.CreateAgentResponse500ResponseBody = void 0;
const zod_1 = require("zod");
/**
 * Internal Server Error
 */
class CreateAgentResponse500ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreateAgentResponse500ResponseBody";
    }
}
exports.CreateAgentResponse500ResponseBody = CreateAgentResponse500ResponseBody;
/**
 * Unprocessable Content
 */
class CreateAgentResponse422ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreateAgentResponse422ResponseBody";
    }
}
exports.CreateAgentResponse422ResponseBody = CreateAgentResponse422ResponseBody;
/**
 * Unauthorized
 */
class CreateAgentResponseResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreateAgentResponseResponseBody";
    }
}
exports.CreateAgentResponseResponseBody = CreateAgentResponseResponseBody;
/**
 * Bad Request
 */
class CreateAgentResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreateAgentResponseBody";
    }
}
exports.CreateAgentResponseBody = CreateAgentResponseBody;
/** @internal */
var CreateAgentResponse500ResponseBody$;
(function (CreateAgentResponse500ResponseBody$) {
    CreateAgentResponse500ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreateAgentResponse500ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreateAgentResponse500ResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreateAgentResponse500ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreateAgentResponse500ResponseBody$ || (exports.CreateAgentResponse500ResponseBody$ = CreateAgentResponse500ResponseBody$ = {}));
/** @internal */
var CreateAgentResponse422ResponseBody$;
(function (CreateAgentResponse422ResponseBody$) {
    CreateAgentResponse422ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreateAgentResponse422ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreateAgentResponse422ResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreateAgentResponse422ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreateAgentResponse422ResponseBody$ || (exports.CreateAgentResponse422ResponseBody$ = CreateAgentResponse422ResponseBody$ = {}));
/** @internal */
var CreateAgentResponseResponseBody$;
(function (CreateAgentResponseResponseBody$) {
    CreateAgentResponseResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreateAgentResponseResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreateAgentResponseResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreateAgentResponseResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreateAgentResponseResponseBody$ || (exports.CreateAgentResponseResponseBody$ = CreateAgentResponseResponseBody$ = {}));
/** @internal */
var CreateAgentResponseBody$;
(function (CreateAgentResponseBody$) {
    CreateAgentResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreateAgentResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreateAgentResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreateAgentResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreateAgentResponseBody$ || (exports.CreateAgentResponseBody$ = CreateAgentResponseBody$ = {}));
