// retell-sdk/models/errors/createphonecall.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePhoneCallResponseBody$ = exports.CreatePhoneCallResponseResponseBody$ = exports.CreatePhoneCallResponse402ResponseBody$ = exports.CreatePhoneCallResponse422ResponseBody$ = exports.CreatePhoneCallResponse429ResponseBody$ = exports.CreatePhoneCallResponse500ResponseBody$ = exports.CreatePhoneCallResponseBody = exports.CreatePhoneCallResponseResponseBody = exports.CreatePhoneCallResponse402ResponseBody = exports.CreatePhoneCallResponse422ResponseBody = exports.CreatePhoneCallResponse429ResponseBody = exports.CreatePhoneCallResponse500ResponseBody = void 0;
const zod_1 = require("zod");
/**
 * Internal Server Error
 */
class CreatePhoneCallResponse500ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreatePhoneCallResponse500ResponseBody";
    }
}
exports.CreatePhoneCallResponse500ResponseBody = CreatePhoneCallResponse500ResponseBody;
/**
 * Too Many Requests
 */
class CreatePhoneCallResponse429ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreatePhoneCallResponse429ResponseBody";
    }
}
exports.CreatePhoneCallResponse429ResponseBody = CreatePhoneCallResponse429ResponseBody;
/**
 * Unprocessable Content
 */
class CreatePhoneCallResponse422ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreatePhoneCallResponse422ResponseBody";
    }
}
exports.CreatePhoneCallResponse422ResponseBody = CreatePhoneCallResponse422ResponseBody;
/**
 * Payment Required
 */
class CreatePhoneCallResponse402ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreatePhoneCallResponse402ResponseBody";
    }
}
exports.CreatePhoneCallResponse402ResponseBody = CreatePhoneCallResponse402ResponseBody;
/**
 * Unauthorized
 */
class CreatePhoneCallResponseResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreatePhoneCallResponseResponseBody";
    }
}
exports.CreatePhoneCallResponseResponseBody = CreatePhoneCallResponseResponseBody;
/**
 * Bad Request
 */
class CreatePhoneCallResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "CreatePhoneCallResponseBody";
    }
}
exports.CreatePhoneCallResponseBody = CreatePhoneCallResponseBody;
/** @internal */
var CreatePhoneCallResponse500ResponseBody$;
(function (CreatePhoneCallResponse500ResponseBody$) {
    CreatePhoneCallResponse500ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreatePhoneCallResponse500ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreatePhoneCallResponse500ResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreatePhoneCallResponse500ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreatePhoneCallResponse500ResponseBody$ || (exports.CreatePhoneCallResponse500ResponseBody$ = CreatePhoneCallResponse500ResponseBody$ = {}));
/** @internal */
var CreatePhoneCallResponse429ResponseBody$;
(function (CreatePhoneCallResponse429ResponseBody$) {
    CreatePhoneCallResponse429ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreatePhoneCallResponse429ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreatePhoneCallResponse429ResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreatePhoneCallResponse429ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreatePhoneCallResponse429ResponseBody$ || (exports.CreatePhoneCallResponse429ResponseBody$ = CreatePhoneCallResponse429ResponseBody$ = {}));
/** @internal */
var CreatePhoneCallResponse422ResponseBody$;
(function (CreatePhoneCallResponse422ResponseBody$) {
    CreatePhoneCallResponse422ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreatePhoneCallResponse422ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreatePhoneCallResponse422ResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreatePhoneCallResponse422ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreatePhoneCallResponse422ResponseBody$ || (exports.CreatePhoneCallResponse422ResponseBody$ = CreatePhoneCallResponse422ResponseBody$ = {}));
/** @internal */
var CreatePhoneCallResponse402ResponseBody$;
(function (CreatePhoneCallResponse402ResponseBody$) {
    CreatePhoneCallResponse402ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreatePhoneCallResponse402ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreatePhoneCallResponse402ResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreatePhoneCallResponse402ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreatePhoneCallResponse402ResponseBody$ || (exports.CreatePhoneCallResponse402ResponseBody$ = CreatePhoneCallResponse402ResponseBody$ = {}));
/** @internal */
var CreatePhoneCallResponseResponseBody$;
(function (CreatePhoneCallResponseResponseBody$) {
    CreatePhoneCallResponseResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreatePhoneCallResponseResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreatePhoneCallResponseResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreatePhoneCallResponseResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreatePhoneCallResponseResponseBody$ || (exports.CreatePhoneCallResponseResponseBody$ = CreatePhoneCallResponseResponseBody$ = {}));
/** @internal */
var CreatePhoneCallResponseBody$;
(function (CreatePhoneCallResponseBody$) {
    CreatePhoneCallResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new CreatePhoneCallResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    CreatePhoneCallResponseBody$.outboundSchema = zod_1.z
        .instanceof(CreatePhoneCallResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(CreatePhoneCallResponseBody$ || (exports.CreatePhoneCallResponseBody$ = CreatePhoneCallResponseBody$ = {}));
