// retell-sdk/models/errors/deleteagent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteAgentResponseBody$ = exports.DeleteAgentResponseResponseBody$ = exports.DeleteAgentResponse422ResponseBody$ = exports.DeleteAgentResponse500ResponseBody$ = exports.DeleteAgentResponseBody = exports.DeleteAgentResponseResponseBody = exports.DeleteAgentResponse422ResponseBody = exports.DeleteAgentResponse500ResponseBody = void 0;
const zod_1 = require("zod");
/**
 * Internal Server Error
 */
class DeleteAgentResponse500ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "DeleteAgentResponse500ResponseBody";
    }
}
exports.DeleteAgentResponse500ResponseBody = DeleteAgentResponse500ResponseBody;
/**
 * Unprocessable Content
 */
class DeleteAgentResponse422ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "DeleteAgentResponse422ResponseBody";
    }
}
exports.DeleteAgentResponse422ResponseBody = DeleteAgentResponse422ResponseBody;
/**
 * Unauthorized
 */
class DeleteAgentResponseResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "DeleteAgentResponseResponseBody";
    }
}
exports.DeleteAgentResponseResponseBody = DeleteAgentResponseResponseBody;
/**
 * Bad Request
 */
class DeleteAgentResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "DeleteAgentResponseBody";
    }
}
exports.DeleteAgentResponseBody = DeleteAgentResponseBody;
/** @internal */
var DeleteAgentResponse500ResponseBody$;
(function (DeleteAgentResponse500ResponseBody$) {
    DeleteAgentResponse500ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new DeleteAgentResponse500ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    DeleteAgentResponse500ResponseBody$.outboundSchema = zod_1.z
        .instanceof(DeleteAgentResponse500ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(DeleteAgentResponse500ResponseBody$ || (exports.DeleteAgentResponse500ResponseBody$ = DeleteAgentResponse500ResponseBody$ = {}));
/** @internal */
var DeleteAgentResponse422ResponseBody$;
(function (DeleteAgentResponse422ResponseBody$) {
    DeleteAgentResponse422ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new DeleteAgentResponse422ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    DeleteAgentResponse422ResponseBody$.outboundSchema = zod_1.z
        .instanceof(DeleteAgentResponse422ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(DeleteAgentResponse422ResponseBody$ || (exports.DeleteAgentResponse422ResponseBody$ = DeleteAgentResponse422ResponseBody$ = {}));
/** @internal */
var DeleteAgentResponseResponseBody$;
(function (DeleteAgentResponseResponseBody$) {
    DeleteAgentResponseResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new DeleteAgentResponseResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    DeleteAgentResponseResponseBody$.outboundSchema = zod_1.z
        .instanceof(DeleteAgentResponseResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(DeleteAgentResponseResponseBody$ || (exports.DeleteAgentResponseResponseBody$ = DeleteAgentResponseResponseBody$ = {}));
/** @internal */
var DeleteAgentResponseBody$;
(function (DeleteAgentResponseBody$) {
    DeleteAgentResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new DeleteAgentResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    DeleteAgentResponseBody$.outboundSchema = zod_1.z
        .instanceof(DeleteAgentResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(DeleteAgentResponseBody$ || (exports.DeleteAgentResponseBody$ = DeleteAgentResponseBody$ = {}));
