// retell-sdk/models/errors/deletephonenumber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeletePhoneNumberResponseBody$ = exports.DeletePhoneNumberResponseResponseBody$ = exports.DeletePhoneNumberResponse422ResponseBody$ = exports.DeletePhoneNumberResponse500ResponseBody$ = exports.DeletePhoneNumberResponseBody = exports.DeletePhoneNumberResponseResponseBody = exports.DeletePhoneNumberResponse422ResponseBody = exports.DeletePhoneNumberResponse500ResponseBody = void 0;
const zod_1 = require("zod");
/**
 * Internal Server Error
 */
class DeletePhoneNumberResponse500ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "DeletePhoneNumberResponse500ResponseBody";
    }
}
exports.DeletePhoneNumberResponse500ResponseBody = DeletePhoneNumberResponse500ResponseBody;
/**
 * Unprocessable Content
 */
class DeletePhoneNumberResponse422ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "DeletePhoneNumberResponse422ResponseBody";
    }
}
exports.DeletePhoneNumberResponse422ResponseBody = DeletePhoneNumberResponse422ResponseBody;
/**
 * Unauthorized
 */
class DeletePhoneNumberResponseResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "DeletePhoneNumberResponseResponseBody";
    }
}
exports.DeletePhoneNumberResponseResponseBody = DeletePhoneNumberResponseResponseBody;
/**
 * Bad Request
 */
class DeletePhoneNumberResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "DeletePhoneNumberResponseBody";
    }
}
exports.DeletePhoneNumberResponseBody = DeletePhoneNumberResponseBody;
/** @internal */
var DeletePhoneNumberResponse500ResponseBody$;
(function (DeletePhoneNumberResponse500ResponseBody$) {
    DeletePhoneNumberResponse500ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new DeletePhoneNumberResponse500ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    DeletePhoneNumberResponse500ResponseBody$.outboundSchema = zod_1.z
        .instanceof(DeletePhoneNumberResponse500ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(DeletePhoneNumberResponse500ResponseBody$ || (exports.DeletePhoneNumberResponse500ResponseBody$ = DeletePhoneNumberResponse500ResponseBody$ = {}));
/** @internal */
var DeletePhoneNumberResponse422ResponseBody$;
(function (DeletePhoneNumberResponse422ResponseBody$) {
    DeletePhoneNumberResponse422ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new DeletePhoneNumberResponse422ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    DeletePhoneNumberResponse422ResponseBody$.outboundSchema = zod_1.z
        .instanceof(DeletePhoneNumberResponse422ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(DeletePhoneNumberResponse422ResponseBody$ || (exports.DeletePhoneNumberResponse422ResponseBody$ = DeletePhoneNumberResponse422ResponseBody$ = {}));
/** @internal */
var DeletePhoneNumberResponseResponseBody$;
(function (DeletePhoneNumberResponseResponseBody$) {
    DeletePhoneNumberResponseResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new DeletePhoneNumberResponseResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    DeletePhoneNumberResponseResponseBody$.outboundSchema = zod_1.z
        .instanceof(DeletePhoneNumberResponseResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(DeletePhoneNumberResponseResponseBody$ || (exports.DeletePhoneNumberResponseResponseBody$ = DeletePhoneNumberResponseResponseBody$ = {}));
/** @internal */
var DeletePhoneNumberResponseBody$;
(function (DeletePhoneNumberResponseBody$) {
    DeletePhoneNumberResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new DeletePhoneNumberResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    DeletePhoneNumberResponseBody$.outboundSchema = zod_1.z
        .instanceof(DeletePhoneNumberResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(DeletePhoneNumberResponseBody$ || (exports.DeletePhoneNumberResponseBody$ = DeletePhoneNumberResponseBody$ = {}));
