// retell-sdk/models/errors/getagent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAgentResponseBody$ = exports.GetAgentResponseResponseBody$ = exports.GetAgentResponse422ResponseBody$ = exports.GetAgentResponse500ResponseBody$ = exports.GetAgentResponseBody = exports.GetAgentResponseResponseBody = exports.GetAgentResponse422ResponseBody = exports.GetAgentResponse500ResponseBody = void 0;
const zod_1 = require("zod");
/**
 * Internal Server Error
 */
class GetAgentResponse500ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "GetAgentResponse500ResponseBody";
    }
}
exports.GetAgentResponse500ResponseBody = GetAgentResponse500ResponseBody;
/**
 * Unprocessable Content
 */
class GetAgentResponse422ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "GetAgentResponse422ResponseBody";
    }
}
exports.GetAgentResponse422ResponseBody = GetAgentResponse422ResponseBody;
/**
 * Unauthorized
 */
class GetAgentResponseResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "GetAgentResponseResponseBody";
    }
}
exports.GetAgentResponseResponseBody = GetAgentResponseResponseBody;
/**
 * Bad Request
 */
class GetAgentResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "GetAgentResponseBody";
    }
}
exports.GetAgentResponseBody = GetAgentResponseBody;
/** @internal */
var GetAgentResponse500ResponseBody$;
(function (GetAgentResponse500ResponseBody$) {
    GetAgentResponse500ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new GetAgentResponse500ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    GetAgentResponse500ResponseBody$.outboundSchema = zod_1.z
        .instanceof(GetAgentResponse500ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(GetAgentResponse500ResponseBody$ || (exports.GetAgentResponse500ResponseBody$ = GetAgentResponse500ResponseBody$ = {}));
/** @internal */
var GetAgentResponse422ResponseBody$;
(function (GetAgentResponse422ResponseBody$) {
    GetAgentResponse422ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new GetAgentResponse422ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    GetAgentResponse422ResponseBody$.outboundSchema = zod_1.z
        .instanceof(GetAgentResponse422ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(GetAgentResponse422ResponseBody$ || (exports.GetAgentResponse422ResponseBody$ = GetAgentResponse422ResponseBody$ = {}));
/** @internal */
var GetAgentResponseResponseBody$;
(function (GetAgentResponseResponseBody$) {
    GetAgentResponseResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new GetAgentResponseResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    GetAgentResponseResponseBody$.outboundSchema = zod_1.z
        .instanceof(GetAgentResponseResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(GetAgentResponseResponseBody$ || (exports.GetAgentResponseResponseBody$ = GetAgentResponseResponseBody$ = {}));
/** @internal */
var GetAgentResponseBody$;
(function (GetAgentResponseBody$) {
    GetAgentResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new GetAgentResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    GetAgentResponseBody$.outboundSchema = zod_1.z
        .instanceof(GetAgentResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(GetAgentResponseBody$ || (exports.GetAgentResponseBody$ = GetAgentResponseBody$ = {}));
