// retell-sdk/models/errors/getphonenumber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetPhoneNumberResponseBody$ = exports.GetPhoneNumberResponseResponseBody$ = exports.GetPhoneNumberResponse422ResponseBody$ = exports.GetPhoneNumberResponse500ResponseBody$ = exports.GetPhoneNumberResponseBody = exports.GetPhoneNumberResponseResponseBody = exports.GetPhoneNumberResponse422ResponseBody = exports.GetPhoneNumberResponse500ResponseBody = void 0;
const zod_1 = require("zod");
/**
 * Internal Server Error
 */
class GetPhoneNumberResponse500ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "GetPhoneNumberResponse500ResponseBody";
    }
}
exports.GetPhoneNumberResponse500ResponseBody = GetPhoneNumberResponse500ResponseBody;
/**
 * Unprocessable Content
 */
class GetPhoneNumberResponse422ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "GetPhoneNumberResponse422ResponseBody";
    }
}
exports.GetPhoneNumberResponse422ResponseBody = GetPhoneNumberResponse422ResponseBody;
/**
 * Unauthorized
 */
class GetPhoneNumberResponseResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "GetPhoneNumberResponseResponseBody";
    }
}
exports.GetPhoneNumberResponseResponseBody = GetPhoneNumberResponseResponseBody;
/**
 * Bad Request
 */
class GetPhoneNumberResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "GetPhoneNumberResponseBody";
    }
}
exports.GetPhoneNumberResponseBody = GetPhoneNumberResponseBody;
/** @internal */
var GetPhoneNumberResponse500ResponseBody$;
(function (GetPhoneNumberResponse500ResponseBody$) {
    GetPhoneNumberResponse500ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new GetPhoneNumberResponse500ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    GetPhoneNumberResponse500ResponseBody$.outboundSchema = zod_1.z
        .instanceof(GetPhoneNumberResponse500ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(GetPhoneNumberResponse500ResponseBody$ || (exports.GetPhoneNumberResponse500ResponseBody$ = GetPhoneNumberResponse500ResponseBody$ = {}));
/** @internal */
var GetPhoneNumberResponse422ResponseBody$;
(function (GetPhoneNumberResponse422ResponseBody$) {
    GetPhoneNumberResponse422ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new GetPhoneNumberResponse422ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    GetPhoneNumberResponse422ResponseBody$.outboundSchema = zod_1.z
        .instanceof(GetPhoneNumberResponse422ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(GetPhoneNumberResponse422ResponseBody$ || (exports.GetPhoneNumberResponse422ResponseBody$ = GetPhoneNumberResponse422ResponseBody$ = {}));
/** @internal */
var GetPhoneNumberResponseResponseBody$;
(function (GetPhoneNumberResponseResponseBody$) {
    GetPhoneNumberResponseResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new GetPhoneNumberResponseResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    GetPhoneNumberResponseResponseBody$.outboundSchema = zod_1.z
        .instanceof(GetPhoneNumberResponseResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(GetPhoneNumberResponseResponseBody$ || (exports.GetPhoneNumberResponseResponseBody$ = GetPhoneNumberResponseResponseBody$ = {}));
/** @internal */
var GetPhoneNumberResponseBody$;
(function (GetPhoneNumberResponseBody$) {
    GetPhoneNumberResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new GetPhoneNumberResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    GetPhoneNumberResponseBody$.outboundSchema = zod_1.z
        .instanceof(GetPhoneNumberResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(GetPhoneNumberResponseBody$ || (exports.GetPhoneNumberResponseBody$ = GetPhoneNumberResponseBody$ = {}));
