// retell-sdk/models/errors/listphonenumbers.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListPhoneNumbersResponseBody$ = exports.ListPhoneNumbersResponseResponseBody$ = exports.ListPhoneNumbersResponse500ResponseBody$ = exports.ListPhoneNumbersResponseBody = exports.ListPhoneNumbersResponseResponseBody = exports.ListPhoneNumbersResponse500ResponseBody = void 0;
const zod_1 = require("zod");
/**
 * Internal Server Error
 */
class ListPhoneNumbersResponse500ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "ListPhoneNumbersResponse500ResponseBody";
    }
}
exports.ListPhoneNumbersResponse500ResponseBody = ListPhoneNumbersResponse500ResponseBody;
/**
 * Unauthorized
 */
class ListPhoneNumbersResponseResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "ListPhoneNumbersResponseResponseBody";
    }
}
exports.ListPhoneNumbersResponseResponseBody = ListPhoneNumbersResponseResponseBody;
/**
 * Bad Request
 */
class ListPhoneNumbersResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "ListPhoneNumbersResponseBody";
    }
}
exports.ListPhoneNumbersResponseBody = ListPhoneNumbersResponseBody;
/** @internal */
var ListPhoneNumbersResponse500ResponseBody$;
(function (ListPhoneNumbersResponse500ResponseBody$) {
    ListPhoneNumbersResponse500ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new ListPhoneNumbersResponse500ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    ListPhoneNumbersResponse500ResponseBody$.outboundSchema = zod_1.z
        .instanceof(ListPhoneNumbersResponse500ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(ListPhoneNumbersResponse500ResponseBody$ || (exports.ListPhoneNumbersResponse500ResponseBody$ = ListPhoneNumbersResponse500ResponseBody$ = {}));
/** @internal */
var ListPhoneNumbersResponseResponseBody$;
(function (ListPhoneNumbersResponseResponseBody$) {
    ListPhoneNumbersResponseResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new ListPhoneNumbersResponseResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    ListPhoneNumbersResponseResponseBody$.outboundSchema = zod_1.z
        .instanceof(ListPhoneNumbersResponseResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(ListPhoneNumbersResponseResponseBody$ || (exports.ListPhoneNumbersResponseResponseBody$ = ListPhoneNumbersResponseResponseBody$ = {}));
/** @internal */
var ListPhoneNumbersResponseBody$;
(function (ListPhoneNumbersResponseBody$) {
    ListPhoneNumbersResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new ListPhoneNumbersResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    ListPhoneNumbersResponseBody$.outboundSchema = zod_1.z
        .instanceof(ListPhoneNumbersResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(ListPhoneNumbersResponseBody$ || (exports.ListPhoneNumbersResponseBody$ = ListPhoneNumbersResponseBody$ = {}));
