// retell-sdk/models/errors/registercall.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegisterCallResponseBody$ = exports.RegisterCallResponseResponseBody$ = exports.RegisterCallResponse402ResponseBody$ = exports.RegisterCallResponse422ResponseBody$ = exports.RegisterCallResponse429ResponseBody$ = exports.RegisterCallResponse500ResponseBody$ = exports.RegisterCallResponseBody = exports.RegisterCallResponseResponseBody = exports.RegisterCallResponse402ResponseBody = exports.RegisterCallResponse422ResponseBody = exports.RegisterCallResponse429ResponseBody = exports.RegisterCallResponse500ResponseBody = void 0;
const zod_1 = require("zod");
/**
 * Internal Server Error
 */
class RegisterCallResponse500ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "RegisterCallResponse500ResponseBody";
    }
}
exports.RegisterCallResponse500ResponseBody = RegisterCallResponse500ResponseBody;
/**
 * Too Many Requests
 */
class RegisterCallResponse429ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "RegisterCallResponse429ResponseBody";
    }
}
exports.RegisterCallResponse429ResponseBody = RegisterCallResponse429ResponseBody;
/**
 * Unprocessable Content
 */
class RegisterCallResponse422ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "RegisterCallResponse422ResponseBody";
    }
}
exports.RegisterCallResponse422ResponseBody = RegisterCallResponse422ResponseBody;
/**
 * Payment Required
 */
class RegisterCallResponse402ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "RegisterCallResponse402ResponseBody";
    }
}
exports.RegisterCallResponse402ResponseBody = RegisterCallResponse402ResponseBody;
/**
 * Unauthorized
 */
class RegisterCallResponseResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "RegisterCallResponseResponseBody";
    }
}
exports.RegisterCallResponseResponseBody = RegisterCallResponseResponseBody;
/**
 * Bad Request
 */
class RegisterCallResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "RegisterCallResponseBody";
    }
}
exports.RegisterCallResponseBody = RegisterCallResponseBody;
/** @internal */
var RegisterCallResponse500ResponseBody$;
(function (RegisterCallResponse500ResponseBody$) {
    RegisterCallResponse500ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new RegisterCallResponse500ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    RegisterCallResponse500ResponseBody$.outboundSchema = zod_1.z
        .instanceof(RegisterCallResponse500ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(RegisterCallResponse500ResponseBody$ || (exports.RegisterCallResponse500ResponseBody$ = RegisterCallResponse500ResponseBody$ = {}));
/** @internal */
var RegisterCallResponse429ResponseBody$;
(function (RegisterCallResponse429ResponseBody$) {
    RegisterCallResponse429ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new RegisterCallResponse429ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    RegisterCallResponse429ResponseBody$.outboundSchema = zod_1.z
        .instanceof(RegisterCallResponse429ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(RegisterCallResponse429ResponseBody$ || (exports.RegisterCallResponse429ResponseBody$ = RegisterCallResponse429ResponseBody$ = {}));
/** @internal */
var RegisterCallResponse422ResponseBody$;
(function (RegisterCallResponse422ResponseBody$) {
    RegisterCallResponse422ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new RegisterCallResponse422ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    RegisterCallResponse422ResponseBody$.outboundSchema = zod_1.z
        .instanceof(RegisterCallResponse422ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(RegisterCallResponse422ResponseBody$ || (exports.RegisterCallResponse422ResponseBody$ = RegisterCallResponse422ResponseBody$ = {}));
/** @internal */
var RegisterCallResponse402ResponseBody$;
(function (RegisterCallResponse402ResponseBody$) {
    RegisterCallResponse402ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new RegisterCallResponse402ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    RegisterCallResponse402ResponseBody$.outboundSchema = zod_1.z
        .instanceof(RegisterCallResponse402ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(RegisterCallResponse402ResponseBody$ || (exports.RegisterCallResponse402ResponseBody$ = RegisterCallResponse402ResponseBody$ = {}));
/** @internal */
var RegisterCallResponseResponseBody$;
(function (RegisterCallResponseResponseBody$) {
    RegisterCallResponseResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new RegisterCallResponseResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    RegisterCallResponseResponseBody$.outboundSchema = zod_1.z
        .instanceof(RegisterCallResponseResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(RegisterCallResponseResponseBody$ || (exports.RegisterCallResponseResponseBody$ = RegisterCallResponseResponseBody$ = {}));
/** @internal */
var RegisterCallResponseBody$;
(function (RegisterCallResponseBody$) {
    RegisterCallResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new RegisterCallResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    RegisterCallResponseBody$.outboundSchema = zod_1.z
        .instanceof(RegisterCallResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(RegisterCallResponseBody$ || (exports.RegisterCallResponseBody$ = RegisterCallResponseBody$ = {}));
