// retell-sdk/models/errors/updateagent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAgentResponseBody$ = exports.UpdateAgentResponseResponseBody$ = exports.UpdateAgentResponse422ResponseBody$ = exports.UpdateAgentResponse500ResponseBody$ = exports.UpdateAgentResponseBody = exports.UpdateAgentResponseResponseBody = exports.UpdateAgentResponse422ResponseBody = exports.UpdateAgentResponse500ResponseBody = void 0;
const zod_1 = require("zod");
/**
 * Internal Server Error
 */
class UpdateAgentResponse500ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "UpdateAgentResponse500ResponseBody";
    }
}
exports.UpdateAgentResponse500ResponseBody = UpdateAgentResponse500ResponseBody;
/**
 * Unprocessable Content
 */
class UpdateAgentResponse422ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "UpdateAgentResponse422ResponseBody";
    }
}
exports.UpdateAgentResponse422ResponseBody = UpdateAgentResponse422ResponseBody;
/**
 * Unauthorized
 */
class UpdateAgentResponseResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "UpdateAgentResponseResponseBody";
    }
}
exports.UpdateAgentResponseResponseBody = UpdateAgentResponseResponseBody;
/**
 * Bad Request
 */
class UpdateAgentResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "UpdateAgentResponseBody";
    }
}
exports.UpdateAgentResponseBody = UpdateAgentResponseBody;
/** @internal */
var UpdateAgentResponse500ResponseBody$;
(function (UpdateAgentResponse500ResponseBody$) {
    UpdateAgentResponse500ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new UpdateAgentResponse500ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    UpdateAgentResponse500ResponseBody$.outboundSchema = zod_1.z
        .instanceof(UpdateAgentResponse500ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(UpdateAgentResponse500ResponseBody$ || (exports.UpdateAgentResponse500ResponseBody$ = UpdateAgentResponse500ResponseBody$ = {}));
/** @internal */
var UpdateAgentResponse422ResponseBody$;
(function (UpdateAgentResponse422ResponseBody$) {
    UpdateAgentResponse422ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new UpdateAgentResponse422ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    UpdateAgentResponse422ResponseBody$.outboundSchema = zod_1.z
        .instanceof(UpdateAgentResponse422ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(UpdateAgentResponse422ResponseBody$ || (exports.UpdateAgentResponse422ResponseBody$ = UpdateAgentResponse422ResponseBody$ = {}));
/** @internal */
var UpdateAgentResponseResponseBody$;
(function (UpdateAgentResponseResponseBody$) {
    UpdateAgentResponseResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new UpdateAgentResponseResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    UpdateAgentResponseResponseBody$.outboundSchema = zod_1.z
        .instanceof(UpdateAgentResponseResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(UpdateAgentResponseResponseBody$ || (exports.UpdateAgentResponseResponseBody$ = UpdateAgentResponseResponseBody$ = {}));
/** @internal */
var UpdateAgentResponseBody$;
(function (UpdateAgentResponseBody$) {
    UpdateAgentResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new UpdateAgentResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    UpdateAgentResponseBody$.outboundSchema = zod_1.z
        .instanceof(UpdateAgentResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(UpdateAgentResponseBody$ || (exports.UpdateAgentResponseBody$ = UpdateAgentResponseBody$ = {}));
