// retell-sdk/models/errors/updatephoneagent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatePhoneAgentResponseBody$ = exports.UpdatePhoneAgentResponseResponseBody$ = exports.UpdatePhoneAgentResponse422ResponseBody$ = exports.UpdatePhoneAgentResponse500ResponseBody$ = exports.UpdatePhoneAgentResponseBody = exports.UpdatePhoneAgentResponseResponseBody = exports.UpdatePhoneAgentResponse422ResponseBody = exports.UpdatePhoneAgentResponse500ResponseBody = void 0;
const zod_1 = require("zod");
/**
 * Internal Server Error
 */
class UpdatePhoneAgentResponse500ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "UpdatePhoneAgentResponse500ResponseBody";
    }
}
exports.UpdatePhoneAgentResponse500ResponseBody = UpdatePhoneAgentResponse500ResponseBody;
/**
 * Unprocessable Content
 */
class UpdatePhoneAgentResponse422ResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "UpdatePhoneAgentResponse422ResponseBody";
    }
}
exports.UpdatePhoneAgentResponse422ResponseBody = UpdatePhoneAgentResponse422ResponseBody;
/**
 * Unauthorized
 */
class UpdatePhoneAgentResponseResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "UpdatePhoneAgentResponseResponseBody";
    }
}
exports.UpdatePhoneAgentResponseResponseBody = UpdatePhoneAgentResponseResponseBody;
/**
 * Bad Request
 */
class UpdatePhoneAgentResponseBody extends Error {
    constructor(err, options) {
        super("", options);
        this.data$ = err;
        if (err.rawResponse != null) {
            this.rawResponse = err.rawResponse;
        }
        if (err.errorMessage != null) {
            this.errorMessage = err.errorMessage;
        }
        const msg = "message" in err && typeof err.message === "string" ? err.message : "";
        const { rawResponse, ...data } = err;
        const content = JSON.stringify(data);
        this.message = [msg, content].filter(Boolean).join("\n");
        this.name = "UpdatePhoneAgentResponseBody";
    }
}
exports.UpdatePhoneAgentResponseBody = UpdatePhoneAgentResponseBody;
/** @internal */
var UpdatePhoneAgentResponse500ResponseBody$;
(function (UpdatePhoneAgentResponse500ResponseBody$) {
    UpdatePhoneAgentResponse500ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new UpdatePhoneAgentResponse500ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    UpdatePhoneAgentResponse500ResponseBody$.outboundSchema = zod_1.z
        .instanceof(UpdatePhoneAgentResponse500ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(UpdatePhoneAgentResponse500ResponseBody$ || (exports.UpdatePhoneAgentResponse500ResponseBody$ = UpdatePhoneAgentResponse500ResponseBody$ = {}));
/** @internal */
var UpdatePhoneAgentResponse422ResponseBody$;
(function (UpdatePhoneAgentResponse422ResponseBody$) {
    UpdatePhoneAgentResponse422ResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new UpdatePhoneAgentResponse422ResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    UpdatePhoneAgentResponse422ResponseBody$.outboundSchema = zod_1.z
        .instanceof(UpdatePhoneAgentResponse422ResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(UpdatePhoneAgentResponse422ResponseBody$ || (exports.UpdatePhoneAgentResponse422ResponseBody$ = UpdatePhoneAgentResponse422ResponseBody$ = {}));
/** @internal */
var UpdatePhoneAgentResponseResponseBody$;
(function (UpdatePhoneAgentResponseResponseBody$) {
    UpdatePhoneAgentResponseResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new UpdatePhoneAgentResponseResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    UpdatePhoneAgentResponseResponseBody$.outboundSchema = zod_1.z
        .instanceof(UpdatePhoneAgentResponseResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(UpdatePhoneAgentResponseResponseBody$ || (exports.UpdatePhoneAgentResponseResponseBody$ = UpdatePhoneAgentResponseResponseBody$ = {}));
/** @internal */
var UpdatePhoneAgentResponseBody$;
(function (UpdatePhoneAgentResponseBody$) {
    UpdatePhoneAgentResponseBody$.inboundSchema = zod_1.z
        .object({
        RawResponse: zod_1.z.instanceof(Response).optional(),
        error_message: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return new UpdatePhoneAgentResponseBody({
            ...(v.RawResponse === undefined
                ? null
                : { rawResponse: v.RawResponse }),
            ...(v.error_message === undefined
                ? null
                : { errorMessage: v.error_message }),
        });
    });
    UpdatePhoneAgentResponseBody$.outboundSchema = zod_1.z
        .instanceof(UpdatePhoneAgentResponseBody)
        .transform((v) => v.data$)
        .pipe(zod_1.z
        .object({
        rawResponse: zod_1.z
            .instanceof(Response)
            .transform(() => {
            throw new Error("Response cannot be serialized");
        })
            .optional(),
        errorMessage: zod_1.z.string().optional(),
    })
        .transform((v) => {
        return {
            ...(v.rawResponse === undefined
                ? null
                : { RawResponse: v.rawResponse }),
            ...(v.errorMessage === undefined
                ? null
                : { error_message: v.errorMessage }),
        };
    }));
})(UpdatePhoneAgentResponseBody$ || (exports.UpdatePhoneAgentResponseBody$ = UpdatePhoneAgentResponseBody$ = {}));
