// retell-sdk/sdk/liveClient.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.convertFloat32ToUint8 = exports.convertUint8ToFloat32 = exports.LiveClient = void 0;
const eventemitter3_1 = require("eventemitter3");
const isomorphic_ws_1 = __importDefault(require("isomorphic-ws"));
const baseEndpoint = "wss://api.re-tell.ai";
class LiveClient extends eventemitter3_1.EventEmitter {
    constructor(apiKey, input) {
        super();
        let endpoint = baseEndpoint +
            "/create-web-call?api_key=" +
            apiKey +
            "&agent_id=" +
            input.agentId;
        if (input.sampleRate != null) {
            endpoint += "&sample_rate=" + input.sampleRate;
        }
        if (input.agentPromptParams != null) {
            endpoint +=
                "&agent_prompt_params=" +
                    encodeURIComponent(JSON.stringify(input.agentPromptParams));
        }
        this.ws = new isomorphic_ws_1.default(endpoint);
        this.ws.binaryType = "arraybuffer";
    }
    waitForReady() {
        return new Promise((resolve, reject) => {
            const onError = (error) => {
                reject(error); // Reject on error
            };
            this.ws.onerror = onError;
            const onClose = (event) => {
                reject("websocket closed before ready with code: " +
                    event.code +
                    ", reason: " +
                    event.reason);
            };
            this.ws.onclose = onClose;
            const onMessage = (event) => {
                try {
                    const data = JSON.parse(event.data);
                    if (data.status === "ready") {
                        // Remove all registered listeners for cleaner future processing.
                        this.ws.removeEventListener("error", onError);
                        this.ws.removeEventListener("close", onClose);
                        this.ws.removeEventListener("message", onMessage);
                        // Emit audio
                        this.ws.onmessage = (event) => {
                            const audio = event.data;
                            this.emit("audio", new Uint8Array(audio));
                        };
                        this.ws.onclose = (event) => {
                            this.emit("close", event);
                        };
                        this.call = data.call;
                        resolve(); // Resolve when the ready message is received
                    }
                }
                catch (error) {
                    // Handle JSON parsing error
                    reject("malformed ready event.");
                }
            };
            this.ws.onmessage = onMessage;
        });
    }
    send(audio) {
        if (this.ws.readyState === 1)
            this.ws.send(audio);
    }
    close() {
        this.ws.close();
    }
    getCall() {
        return this.call;
    }
}
exports.LiveClient = LiveClient;
function convertUint8ToFloat32(array) {
    const targetArray = new Float32Array(array.byteLength / 2);
    // A DataView is used to read our 16-bit little-endian samples out of the Uint8Array buffer
    const sourceDataView = new DataView(array.buffer);
    // Loop through, get values, and divide by 32,768
    for (let i = 0; i < targetArray.length; i++) {
        targetArray[i] = sourceDataView.getInt16(i * 2, true) / Math.pow(2, 16 - 1);
    }
    return targetArray;
}
exports.convertUint8ToFloat32 = convertUint8ToFloat32;
function convertFloat32ToUint8(array) {
    const buffer = new ArrayBuffer(array.length * 2);
    const view = new DataView(buffer);
    for (let i = 0; i < array.length; i++) {
        const value = array[i] * 32768;
        view.setInt16(i * 2, value, true); // true for little-endian
    }
    return new Uint8Array(buffer);
}
exports.convertFloat32ToUint8 = convertFloat32ToUint8;
